src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1222,146 +1221,44 @@ private IArrayInitializerOperation CreateBoundArrayInitializationOperation(Bound
-                    OperationFactory.IdentityConversion,
-                    _semanticModel,
-                    syntax,
-                    isImplicit: true);
-                return new ObjectCreationOperation(
-                    spanConstructor,
-                    initializer: null,
-                    ImmutableArray.Create(constructorArgument),
-                    _semanticModel,
-                    syntax,
-                    collectionType,
-                    constantValue: null,
-                    isImplicit);
+                var builder = ArrayBuilder<IOperation>.GetInstance(elements.Length);
+                foreach (var element in elements)
+                {
+                    var child = createChild(element);
+                    if (child is { })
+                    {
+                        builder.Add(child);
+                    }
+                }
+                return builder.ToImmutableAndFree();
             }
-        }
 
-        private IOperation CreateBoundCollectionInitializerCollectionLiteralExpression(BoundCollectionLiteralExpression boundCollectionLiteralExpression)
-        {
-            SyntaxNode syntax = boundCollectionLiteralExpression.Syntax;
-            ITypeSymbol? collectionType = boundCollectionLiteralExpression.GetPublicTypeSymbol();
-            bool isImplicit = boundCollectionLiteralExpression.WasCompilerGenerated;
-            var initializer = new ObjectOrCollectionInitializerOperation(
-                CreateFromArray<BoundExpression, IOperation>(boundCollectionLiteralExpression.Elements),
-                _semanticModel,
-                syntax,
-                collectionType,
-                isImplicit: true);
-            if (collectionType is ITypeParameterSymbol)
+            IOperation? createChild(BoundExpression expression)
             {
-                return new TypeParameterObjectCreationOperation(initializer, _semanticModel, syntax, collectionType, isImplicit: isImplicit);
-            }
-            else if (boundCollectionLiteralExpression.CollectionCreation is BoundObjectCreationExpression collectionCreation)
-            {
-                return new ObjectCreationOperation(
-                    collectionCreation.Constructor.GetPublicSymbol(),
-                    initializer,
-                    DeriveArguments(collectionCreation),
-                    _semanticModel,
-                    syntax,
-                    collectionType,
-                    constantValue: null,
-                    isImplicit: isImplicit);
-            }
-            else
-            {
-                return new InvalidOperation(ImmutableArray.Create<IOperation>(initializer), _semanticModel, syntax, collectionType, constantValue: null, isImplicit);
+                var element = expression switch
+                {
+                    BoundCollectionElementInitializer initializer => initializer.Arguments.FirstOrDefault(),
```

Comment:
When is this `Default`?

Reply:
This should never be `Default`, but it also didn't seem worth requiring an argument here since we're already allowing other error conditions (such as multiple values, and `null` values).